pkg/odo/util/pushtarget/pushtarget.go

```diff
@@ -0,0 +1,45 @@
+package pushtarget
+
+import (
+	"os"
+
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/preference"
+)
+
+// env variables
+const (
+	// Setting this env to `docker` will enable pushing to docker containers
+	// and will override the setting in the preferences file.
+
+	OdoPushTarget = "ODO_PUSHTARGET"
+)
+
+// IsPushTargetDocker checks if the push target preference has been set to docker
+func IsPushTargetDocker() bool {
+
+	// ODO's push target can be told to use docker by:
+	//		- setting an env variable "ODO_PUSHTARGET=docker" or
```

Comment:
if the user has set both but to opposite values, the env variable will be the first choice?

Reply:
Yeah, my thinking was to have the environment var override what was set in the preferences file (perhaps the user wants to use `ODO_PUSH_TARGET=docker` for a one off `odo push`, without wanting to change the global preferences.